process/headerCheck/headerIntegrityVerifier.go

```diff
@@ -0,0 +1,63 @@
+package headerCheck
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data"
+)
+
+type headerIntegrityVerifier struct {
+	referenceChainID []byte
+}
+
+// NewHeaderIntegrityVerifier returns a new instance of headerIntegrityVerifier
+func NewHeaderIntegrityVerifier(referenceChainID []byte) (*headerIntegrityVerifier, error) {
+	if len(referenceChainID) == 0 {
+		return nil, ErrInvalidReferenceChainID
+	}
+
+	return &headerIntegrityVerifier{
+		referenceChainID: referenceChainID,
+	}, nil
+}
+
+// Verify will check the header's fields such as the chain ID or the software version
+func (h *headerIntegrityVerifier) Verify(hdr data.HeaderHandler) error {
+	err := h.checkSoftwareVersion(hdr)
+	if err != nil {
+		return err
+	}
+
+	return h.checkChainID(hdr)
+}
+
+// CheckSoftwareVersion returns nil if the software version has the correct length
```

Comment:
fix comment ,name of the method starts with lower letter

Reply:
done